src/main/scala/org/virtuslab/yaml/yaml.scala

```diff
@@ -1,17 +1,16 @@
 package org.virtuslab.scala.yaml
 
-import org.virtuslab.yaml.{YamlDecoder, YamlEncoder}
-import org.virtuslab.internal.load.compose.Node
-import org.virtuslab.internal.load.compose.ComposerImpl
 import org.virtuslab.internal.YamlError
+import org.virtuslab.internal.load.compose.ComposerImpl
+import org.virtuslab.internal.load.compose.Node
+import org.virtuslab.yaml.YamlEncoder
+import org.virtuslab.yaml.Util
 
 import scala.deriving._
+import org.virtuslab.internal.load.construct.Construct
 
-def deriveEncoder[T](using m: Mirror.Of[T]) = YamlEncoder.derived
-def deriveDecoder[T](using m: Mirror.Of[T]) = YamlDecoder.derived
-
-def parse(yaml: String): Either[YamlError, Node] = ComposerImpl.compose(yaml)
-extension (str: String)
-  def fromYaml[T](using decoder: YamlDecoder[T]): Either[Any, T] = decoder.from(str)
+inline def deriveEncoder[T](using m: Mirror.Of[T]): YamlEncoder[T] = YamlEncoder.derived[T]
+inline def deriveDecoder[T](using m: Mirror.Of[T]): Construct[T]   = Construct.derived[T]
 
-extension [T](t: T) def asYaml(using encoder: YamlEncoder[T]): String = encoder.toYaml(t)
+extension (str: String) def as[T](using c: Construct[T]): Either[YamlError, T] = Util.as(str)
+extension [T](t: T) def asYaml(using encoder: YamlEncoder[T]): String          = Util.asYaml(t)
```

Comment:
`munit` had a problem with extension methods written at top-level so I moved it to the `Util` object. It's a little boilerplate to repeat `extension (str: String) def as[T]...` 2 times, but at least there is only one implementation of that method